pallets/committee/src/lib.rs

```diff
@@ -450,10 +454,19 @@ pub mod pallet {
 					result.map(|_| ()).map_err(|e| e.error),
 				));
 
-				// TODO: Handle weight used by the dispatch call in weight calculation
+				Ok(())
+			})?;
 
-				Ok(().into())
-			})
+			// Set new `VotingPeriod` if no active proposals left and `PendingVotingPeriod` is some
+			if Proposals::<T>::iter_values().filter(|p| p.status == ProposalStatus::Active).count().is_zero() {
+				if let Some(pending_voting_period) = PendingVotingPeriod::<T>::get() {
+					VotingPeriod::<T>::set(pending_voting_period);
+				}
+			}
```

Comment:
this is not deterministic because there might be no proposal in the voting period so pending voting period won't be set until a new proposal gets closed